### PR TITLE
feat: NVSHAS-9501 support SLSA L3

### PR DESCRIPTION
**Pull request by holyspectral**
1. Move the Jenkins build to GHA as dev.yml and dev-arm64.yml
2. Provide release.yml as SLSA-compliant release build.

### CODE REVIEW

**Comment by flavio** [comment]
From a quick look:
* I would use a `matrix` to merge the arm64 and x86_64 jobs into a single file
* The commit about the log levels doesn't belong to this PR

**Comment by holyspectral** [comment]
@flavio Thanks for the comment!
> I would use a matrix to merge the arm64 and x86_64 jobs into a single file

Yes I agree that will be much simple in the code level if we merge them into one file.  There are two reasons why it's separate right now:

1. Many of our components don't support ARM64 right now, so the errors on ARM64 build shouldn't break AMD64 build. 
2. ARM64 takes more time to build as it's emulated. Currently AMD64 build takes only 2.5 minutes, while ARM64 takes 18 mins.  Having them separate will save my time to do the test.

I will visit this once we setup GitHub-hosted ARM64 runners in neuvector, which should speed up the ARM64 build. 

Also, according to [GitHub blog](https://github.blog/changelog/2024-06-03-actions-arm-based-linux-and-windows-runners-are-now-in-public-beta/), `We expect to begin offering Arm runners for open source and personal accounts by the end of the year.`.  Hope folks can start using this in forks soon.  

> The commit about the log levels doesn't belong to this PR

Oops, it's fixed now.  Thanks for noticing it.

File: build/Dockerfile [hunk 1]
```diff
@@ -1,12 +1,46 @@
...
+FROM micro
+ARG VERSION
+ARG COMMIT
+WORKDIR /
+COPY --from=base /chroot/ /
+COPY --from=builder /src/stage /
+
+RUN cd /usr/bin/ && rm -rf basename chcon chgrp chmod chown chroot cksum dd df dircolors dirname du install install-info join locale localedef mkdir mkfifo mknod mktemp paste pathchk readlink realpath sync smidiff smidump smilink smiquery smistrip smixlate tee tiemout tload top truncate unlink watch
+
+LABEL name="registry-adapter" \
```
Comment (pjbgf) [inline build/Dockerfile:38]: Please note that by using BCI images, you may need to overwrite some specific labels, or you will end-up with labels from the base image instead. [Here](https://github.com/rancher/rancher/pull/46796#issue-2481720724)'s more context.

File: build/Dockerfile [hunk 2]
```diff
@@ -1,12 +1,46 @@
...
+      vendor="SuSE Security" \
```
Comment (pjbgf) [inline build/Dockerfile:39]: Should this not be all capitals these days?

```suggestion
      vendor="SUSE Security" \
```

File: build/Dockerfile [hunk 3]
```diff
@@ -1,12 +1,46 @@
-ARG BASE_IMAGE_TAG
-FROM neuvector/adapter_base:${BASE_IMAGE_TAG}
+#
+# Builder image
+FROM registry.suse.com/bci/golang:1.22.7 AS builder
```
Comment (pjbgf) [inline build/Dockerfile:3]: Either lean on a higher level mutable tag, or add `renovate`/`updatecli` to ensure this is being kept up to date.

```suggestion
FROM registry.suse.com/bci/golang:1.22 AS builder
```

File: build/Dockerfile [hunk 4]
```diff
@@ -1,12 +1,46 @@
...
+    rm -rf /chroot/var/log/
+
+FROM micro
+ARG VERSION
+ARG COMMIT
+WORKDIR /
+COPY --from=base /chroot/ /
+COPY --from=builder /src/stage /
+
+RUN cd /usr/bin/ && rm -rf basename chcon chgrp chmod chown chroot cksum dd df dircolors dirname du install install-info join locale localedef mkdir mkfifo mknod mktemp paste pathchk readlink realpath sync smidiff smidump smilink smiquery smistrip smixlate tee tiemout tload top truncate unlink watch
```
Comment (pjbgf) [inline build/Dockerfile:36]: This creates a new layer on the final image, so although the binaries won't be shipped, they would already be present on higher layers which translate in little to no gains in final image size. An alternative here, is to do this on an intermediate layer, and just copy that over to a final `scratch` image.

File: Makefile [hunk 5]
```diff
@@ -1,26 +1,74 @@
...
+	CHANGES = $(shell git status --porcelain --untracked-files=no)
+	ifneq ($(CHANGES),)
+		DIRTY = -dirty
+	endif
+
+	# Prioritise DRONE_TAG for backwards compatibility. However, the git tag
+	# command should be able to gather the current tag, except when the git
+	# clone operation was done with "--no-tags".
+	ifneq ($(DRONE_TAG),)
+		GIT_TAG = $(DRONE_TAG)
```
Comment (pjbgf) [inline Makefile:26]:
```suggestion
	ifneq ($(GITHUB_TAG),)
		GIT_TAG = $(GITHUB_TAG)
```

File: build/Dockerfile (hunk 4 above)
Comment (holyspectral) [reply build/Dockerfile:36]: yeah this is the code that the current pipeline has.  I will check to see if we should do it with SLSA scope or later.  Thanks for letting us know!

**Comment by holyspectral** [comment]
I will be working on the PR a little more.  Closing it for now to prevent spam. :-)

File: build/Dockerfile (hunk 3 above)
Comment (holyspectral) [reply build/Dockerfile:3]: @jayhuang-suse any concern about this? I don't think Golang would bring breaking changes in minor update, so I'm okay with using 1.22.

File: build/Dockerfile (hunk 1 above)
Comment (holyspectral) [reply build/Dockerfile:38]: Thanks. It's very good to know.

File: Makefile [hunk 6]
```diff
@@ -1,26 +1,74 @@
...
+ifeq ($(VERSION),)
+	# Define VERSION, which is used for image tags or to bake it into the
+	# compiled binary to enable the printing of the application version, 
+	# via the --version flag.
+	CHANGES = $(shell git status --porcelain --untracked-files=no)
+	ifneq ($(CHANGES),)
+		DIRTY = -dirty
+	endif
+
+	# Prioritise DRONE_TAG for backwards compatibility. However, the git tag
```
[the remark this answers is not in the document]
Comment (holyspectral) [reply Makefile:22]: The code comes from cis-operator and I think that's for DRONE only.  I will remove this part.

**Pull request closed without merging**